modules/eks/argocd/github_webhook.tf

```diff
@@ -0,0 +1,74 @@
+# The GitHub webhook can be created with this component, with another component (such as github-webhook), or manually.
+# However, we need to define the value for the webhook secret now with the ArgoCD chart deployment. Store it in SSM for reference.
+#
+# We need to create the webhook with a separate component if we're deploying argocd-repos for multiple GitHub Organizations
+locals {
+  github_webhook_enabled = local.enabled && var.github_webhook_enabled
+  create_github_webhook  = local.github_webhook_enabled && var.create_github_webhook
```

Comment:
What is the difference?

Reply:
`github_webhook_enabled` is whether or not to create a secret value and pass it to the `argo-cd` chart

`create_github_webhook` is whether or not to create a GitHub Webhook for the argocd repos. If you're creating webhooks for multiple organizations and need different github providers, you have to use the `github-webhook` component instead. But you still want to have the same value used for the `argo-cd` chart

> 1. When deploying argocd-repo into different GitHub Organizations, we need to use a different Terraform GitHub provider for each organization. Since Terraform does not support dynamic providers, we have to use multiple components for each provider that we need.
> 2. We can create the GitHub webhook secret value with the eks/argocd component (where it's needed for the chart) and store it in SSM. Then we can use this new component github-webhook to create the webhook in each repo.
> 3. The webhook secret will be the same for each repo connected to the same argocd deployment

https://github.com/cloudposse/terraform-aws-components/blob/46b22591d004c9961daf843abaf49b5ea8e45cdd/modules/eks/argocd/github_webhook.tf#L18-L28